frontend/packages/console-app/src/components/nodes/NodeStatus.tsx

```diff
@@ -1,16 +1,61 @@
 import * as React from 'react';
+import * as _ from 'lodash';
+import ConsumerPopover from '@console/shared/src/components/dashboard/utilization-card/TopConsumerPopover';
 import { Status, SecondaryStatus, getNodeSecondaryStatus } from '@console/shared';
 import { NodeKind } from '@console/internal/module/k8s';
+import { humanizeBinaryBytes, humanizeNumber } from '@console/internal/components/utils';
 import { nodeStatus } from '../../status/node';
+import { PressureQueries, Condition } from '../../queries';
+import './node-status.scss';
+
+const conditionDescriptionMap = Object.freeze({
+  [Condition.DISK_PRESSURE]: 'available memory is low',
+  [Condition.MEM_PRESSURE]: 'available disk capacity is low',
+  [Condition.PID_PRESSURE]: 'CPU is running a large number of processes',
+});
+
+const humanizeMap = Object.freeze({
+  [Condition.DISK_PRESSURE]: humanizeBinaryBytes,
+  [Condition.MEM_PRESSURE]: humanizeBinaryBytes,
+  [Condition.PID_PRESSURE]: humanizeNumber,
+});
+
+const isMonitoredCondition = (condition: Condition): boolean =>
+  [Condition.DISK_PRESSURE, Condition.MEM_PRESSURE, Condition.PID_PRESSURE].includes(condition);
+
+const getDegradedStates = (node: NodeKind): Condition[] => {
+  return node.status.conditions
+    .filter(({ status, type }) => status === 'True' && isMonitoredCondition(type as Condition))
+    .map(({ type }) => type as Condition);
+};
+
+const getPopovers = (items: Condition[], node: string): React.ReactElement[] =>
```

Comment:
I would either make this a real component or just inline it. Helper functions that return React elements seem worse than making a component and isn't really a pattern we should start imo.

Reply:
ack